js/about/aboutActions.js

```diff
@@ -102,6 +102,13 @@ const aboutActions = {
     ipc.send(messages.RESET_SYNC)
   },
 
+  /**
+   * Dispatches a message to trigger a browser restart notification
+   */
+  requireRestart: function () {
```

Comment:
this should send the key and value so the restart handler doesn't prompt for restart if you toggle the setting 2 times in a row. you can also replace PREFS_RESTART in ` js/about/preferences.js` with this.